test.py

```diff
@@ -64,6 +64,26 @@ def test_raises_for_non_csv(self, mock_urlopen):
         with self.assertRaisesRegexp(Exception, err):
             self.stream._api_call(None)
 
+    # Handles unencodable strings
+    @patch('ocr.urllib2.urlopen')
+    def test_hanldes_unencodable_strings(self, mock_urlopen):
+        res_mock = Mock()
+        res_mock.info.side_effect = self.foo
+        # The characters here represent a copywrite symbol
+        res_mock.read.side_effect = lambda: "{'problem_text': 'OHNO!\xc2\xae'}"
+        mock_urlopen.return_value = res_mock
+
+        exception_raised = False
+        try:
+            self.stream.read()
+        except Exception, e:
+            exception_raised = e
+
+        self.assertFalse(exception_raised, exception_raised)
+
+    def foo(self):
```

Comment:
what is foo?? why not response_info?
also, you can use inline lambda as with mock.read.side_effect